### PR TITLE
Make download button <a> + parse url

### PR DESCRIPTION
**Pull request by joeyballentine**
- replace download `<button>` with `<a>`
- add more known hosts
- parse URL for unknown hosts
- Extract components

### CODE REVIEW

File: src/elements/components/download-button.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    resource: Resource;
+};
+
+const getHostFromUrl = (url: string) => {
+    if (url.includes('github')) {
+        return 'GitHub';
+    }
+    if (url.includes('drive.google')) {
+        return 'Google Drive';
+    }
```
Comment (RunDevelopment) [inline src/elements/components/download-button.tsx:15]: I can only see this go wrong long term. Could you parse the URL and look at the domain instead?

Comment (joeyballentine) [reply src/elements/components/download-button.tsx:15]: that's a good point lol..... `www.drive.google.suslink.no/dont-click-this.exe`

File: src/lib/util.ts [hunk 2]
```diff
@@ -79,3 +79,16 @@ export function typedEntries<K extends string, V>(o: Record<K, V>): [K, V][] {
 export function typedKeys<K extends string>(o: Record<K, unknown>): K[] {
     return Object.keys(o) as K[];
 }
+
+export function getColorMode(numberOfChannels: number | string) {
```
Comment (RunDevelopment) [inline src/lib/util.ts:83]: 1. Parsing numbers should not be the responsibility of this function.
2. You only ever call the function with numbers.
```suggestion
export function getColorMode(numberOfChannels: number) {
```

File: src/pages/models/[id].tsx [hunk 3]
```diff
@@ -3,13 +3,13 @@ import Head from 'next/head';
 import Link from 'next/link';
 import { ParsedUrlQuery } from 'querystring';
 import React from 'react';
-import { ReactCompareSlider, ReactCompareSliderHandle, ReactCompareSliderImage } from 'react-compare-slider';
-import { BsCaretLeftFill, BsCaretRightFill } from 'react-icons/bs';
-import { FiExternalLink } from 'react-icons/fi';
+
+import { DownloadButton } from '../../elements/components/download-button';
```
Comment (RunDevelopment) [inline src/pages/models/[id].tsx:7]:
```suggestion
import { DownloadButton } from '../../elements/components/download-button';
```

File: src/lib/util.ts [hunk 4]
```diff
@@ -79,3 +79,16 @@ export function typedEntries<K extends string, V>(o: Record<K, V>): [K, V][] {
 export function typedKeys<K extends string>(o: Record<K, unknown>): K[] {
     return Object.keys(o) as K[];
 }
+
+export function getColorMode(numberOfChannels: number) {
+    switch (Number(numberOfChannels)) {
```
Comment (joeyballentine) [inline src/lib/util.ts:84]:
```suggestion
    switch (numberOfChannels) {
```

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**